### ISSUE TITLE
ssl_cert and ssl_key are required

### ISSUE THREAD

**Issue by gerases** [voxpupuli/puppet-nginx#743]
The code below requires ssl_cert and ssl_key to be defined. But I have an include file that defines those instead :( Any way around this?

```ruby
# Check to see if SSL Certificates are properly defined.
  if ($ssl == true) {
    if ($ssl_cert == undef) or ($ssl_key == undef) {
      fail('nginx: SSL certificate/key (ssl_cert/ssl_key) and/or SSL Private must be defined and exist on the target system(s)')
    }
  }
```

Also, I think the ability not to specify the ciphers would be nice.

**Comment by 3flex** [comment]
Hmm... we could downgrade to a warning or info message instead of a hard fail.

Just wondering why the include file instead of using these parameters? I'm not blocking this just yet, but I want to understand the use case before changing the existing behaviour which is useful for 99% of our users. Warnings or info messages are more easily missed.

**Comment by wyardley** [comment]
#945 now allows keys to have an empty value if set to undef. This has been merged in the most recent release (0.5.0), and should resolve your issue.

**Comment by gerases** [comment]
Awesome, thank you!
